com.unity.netcode.gameobjects/Runtime/SceneManagement/SceneEventData.cs

```diff
@@ -1036,14 +1044,94 @@ private void DeserializeObjectsMovedIntoNewScene(FastBufferReader reader)
                     reader.ReadValueSafe(out networkObjectId);
                     if (!spawnManager.SpawnedObjects.ContainsKey(networkObjectId))
                     {
-                        throw new Exception($"[Object Scene Migration] Trying to synchronize NetworkObjectId ({networkObjectId}) but it no longer exists!");
+                        NetworkLog.LogError($"[Object Scene Migration] Trying to synchronize NetworkObjectId ({networkObjectId}) but it was not spawned or no longer exists!!");
+                        continue;
                     }
+                    // Add NetworkObject scene migration to ObjectsMigratedIntoNewScene dictionary that is processed
+                    //
                     sceneManager.ObjectsMigratedIntoNewScene[sceneHandle].Add(spawnManager.SpawnedObjects[networkObjectId]);
                 }
             }
         }
 
 
+        /// <summary>
+        /// While a client is synchronizing ObjectSceneChanged messages could be received.
+        /// This defers any ObjectSceneChanged message processing to occur after the client
+        /// has completed synchronization to assure the associated NetworkObjects being
+        /// migrated to a new scene are instantiated and spawned.
+        /// </summary>
+        private void DeferObjectsMovedIntoNewScene(FastBufferReader reader)
+        {
+            var sceneManager = m_NetworkManager.SceneManager;
+            var spawnManager = m_NetworkManager.SpawnManager;
+            var numberOfScenes = 0;
+            var sceneHandle = 0;
+            var objectCount = 0;
+            var networkObjectId = (ulong)0;
+
+            var deferredObjectsMovedEvent = new NetworkSceneManager.DeferredObjectsMovedEvent()
+            {
+                ObjectsMigratedTable = new Dictionary<int, List<ulong>>()
```

Comment:
Since these can be created dynamically at any time here, I worry about the GC pressure that might be added if a lot of these come in rapidly. Could this be changed from `Dictionary<int, List<ulong>>` to `NativeHashMap<int, NativeList<ulong>>`? Allocating with `Allocator.Persistent` would be required but that's still better than GC allocation... especially given the potential number of calls to `new List<ulong>()` below.

Reply:
This will only happen under the following conditions:

- A client is going through the initial synchronization process after the client has connected and the connection has been approved.
- While the client is processing the synchronization message (i.e. loading scenes, instantiating NetworkObjects, etc) the client receives and processes a ObjectSceneChanged event.

So, regarding memory allocation it is basically during a period of time where GC will always be at its "highest peak".  Once a client is synchronized this section of code will never get invoked...so in reality the DeferredObjectsMovedEvent structures won't be allocated at any time... only during client synchronization and only if the client receives any ObjectSceneChanged events...which each ObjectSceneChanged event can contain (n) number of scene handles with associated NetworkObject identifiers (i.e. the server batches NetworkObject scene changes together for each frame... so if a user migrates 100 NetworkObjects into a different scene all at once there will be 1 ObjectSceneChanged event that contains the target scene handle and all of the NetworkObject identifiers to migrate into the target scene).